sw/device/lib/testing/sram_ctrl_testutils.c

```diff
@@ -51,8 +51,8 @@ bool sram_ctrl_testutils_read_check_eq(
   return true;
 }
 
-bool sram_ctrl_testutils_read_check_neq(
-    uintptr_t address, const sram_ctrl_testutils_data_t *expected) {
+bool sram_ctrl_testutils_check_data_neq(
+    uintptr_t address, const sram_ctrl_testutils_data_t expected) {
```

Comment:
nit: Can we use `memcmp` instead of this?

Reply:
Ditto.

Reply:
Please review again, I replaced by the `CHECK_ARRAYS_NE` that will produce a nice log in case of unmatch.